chirho/explainable/internals/defaults.py

```diff
@@ -173,3 +174,90 @@ def _soft_neq_independent(support: constraints.independent, v1: T, v2: T, **kwar
     for _ in range(support.reinterpreted_batch_ndims):
         result = torch.sum(result, dim=-1)
     return result
+
+
+class InferSupports(pyro.poutine.messenger.Messenger):
+    """
+    A Pyro Messenger for inferring distribution constraints.
+
+    :param antecedents: (optional) A list or mapping of antecedent variables or constraints.
+    :param witnesses: (optional) A list or mapping of witness variables or constraints.
+    :param consequents: (optional) A list or mapping of consequent variables or constraints.
+
+    :return: An instance of InferSupports with new attributes: ``supports``, ``antecedents``,
+        ``witnesses``, and ``consequents``. ``supports`` is a dictionary mapping variable names
+        to constraints for all variables in the model, while the other attributes are analogous,
+        restricted to variables mentioned in the optional parameters (empty dictionaries if no
+        optional parameters are provided).
+
+    Example:
+
+        >>> def mixed_supports_model():
+        >>>     uniform_var = pyro.sample("uniform_var", dist.Uniform(1, 10))
+        >>>     normal_var = pyro.sample("normal_var", dist.Normal(3, 15))
+        >>> with InferSupports(antecedents=["uniform_var"]) as s:
+        ...      mixed_supports_model()
+        >>> print(s.supports, s.antecedents)
+    """
+
+    supports: MutableMapping[str, pyro.distributions.constraints.Constraint]
+
+    def __init__(
+        self,
+        antecedents: Optional[
```

Comment:
I don't think these arguments are necessary. This handler isn't specific to usage with `SearchForExplanation`, so the distinction between antecedents, witnesses and consequents should just be enforced by code that accesses the results of this handler in its `self.supports`.

Reply:
removed